internal/app/machined/pkg/controllers/runtime/kernel_param_defaults.go

```diff
@@ -120,6 +120,11 @@ func (ctrl *KernelParamDefaultsController) getKernelParams() []*kernel.Param {
 			Key:   "proc.sys.net.ipv4.tcp_keepalive_intvl",
 			Value: "60",
 		},
+		// Consider fwmark for rp_filter routing table lookup.
+		{
+			Key:   "proc.sys.net.ipv4.conf.kubespan.src_valid_mark",
```

Comment:
shouldn't this be `rp_filter` instead of `src_valid_mark` :thinking: ?

Reply:
rp_filter will get overriden. This is what I mean:
```
src_valid_mark - BOOLEAN

        0 - The fwmark of the packet is not included in reverse path route lookup. This allows for asymmetric routing configurations utilizing the fwmark in only one direction, e.g., transparent proxying.

        1 - The fwmark of the packet is included in reverse path route lookup. This permits rp_filter to function when the fwmark is used for routing traffic in both directions.

    This setting also affects the utilization of fmwark when performing source address selection for ICMP replies, or determining addresses stored for the IPOPT_TS_TSANDADDR and IPOPT_RR IP options.

    The max value from conf/{all,interface}/src_valid_mark is used.

    Default value is 0.
```
https://docs.kernel.org/networking/ip-sysctl.html

Reply:
It's not to disable rp_filter (you cannot override someone enabling rp_filter in mode 2 (loose), like current Kube-Router versions), but to make it honor policy-based routing. To avoid potential side effects we only do this on kubespan interface as it's the interface we do our policy-based routing on.

Reply:
okay, that makes sense :+1: